public/app/features/plugins/admin/components/PluginListItem.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { useDisplayMode } from '../state/hooks';
+import { CatalogPlugin, PluginListDisplayMode } from '../types';
+import { PluginListItemRow } from './PluginListItemRow';
+import { PluginListItemCard } from './PluginListItemCard';
+
+export const LOGO_SIZE = '48px';
+
+type Props = {
+  plugin: CatalogPlugin;
+  pathName: string;
+};
+
+export function PluginListItem({ plugin, pathName }: Props) {
+  const { displayMode } = useDisplayMode();
+  const isList = displayMode === PluginListDisplayMode.List;
+
+  if (isList) {
+    return <PluginListItemRow plugin={plugin} pathName={pathName} />;
+  }
+
+  return <PluginListItemCard plugin={plugin} pathName={pathName} />;
+}
+
+// Styles shared between the different type of list items
+export const getStyles = (theme: GrafanaTheme2, displayMode: PluginListDisplayMode) => {
```

Comment:
My initial thought was to split the smaller "shared" parts into small components and share the styling via the components that is being reused in `PluginListItemRow` and `PluginListItemCard`.

But I think this is a good approach that we can stick with!

Reply:
Yea, I was kinda trying to go down that path, but found the components a bit too atomic and small, so stopped. 
I am not fully satisfied with the current way of it either though. Hmm, hmm... 

I can massage it a bit more in that direction if you guys would find that easier to read/maintain?

Reply:
Nah, I think this is good enough for now! I would suggest to stick with this and maybe give it some more love if we need to jump in and add/change something more in a near future.

How does that sound?

Reply:
![55784169](https://user-images.githubusercontent.com/9974811/135273405-5cb5e54e-12e3-4856-8151-11d3f85bd62a.jpg)